Narrativa API: COVID-19 data

ShoiraTa opened the pull request:
## Covid Stats web-app
Selected an API that provides numeric data about a topic that you like and then build the webapp around it. The webapp will has several pages:

- one page with a list of items that could be filtered by some parameters; in the example below, it's a list of metrics that can be filtered by the country (imagine a search field to introduce the country name like Italy, Croatia, etc.). This page should be your homepage.
- one page for the item details; in the example, the detail page for Czech Republic cities with number of views

Video Description https://www.loom.com/share/c59f284ffbfa4a3f9bc8b8210b8d0707

ShoiraTa committed:
Add navbar

ShoiraTa committed:
Add layout

ShoiraTa committed:
Add countries homepage

ShoiraTa committed:
Add regions page

ShoiraTa committed:
Add tests

ShoiraTa committed:
Add styles

ShoiraTa committed:
Updated meta data

ShoiraTa committed:
Add deployment link

Lameck1 commented on src/__tests__/Countries.test.js:
```diff
@@ -0,0 +1,13 @@
+import store from '../redux/store';
+import Countries from '../components/Countries';
+
+describe('Countries', () => {
+  it('renders correctly', () => {
+    const tree = render(<Provider store={store}><Router><Countries /></Router></Provider>);
+    expect(tree).toMatchSnapshot();
+  });
+});
```
- [ ] You have done well including snapshot tests 👏🏼. Let's take a look at the testing requirements though:
  
| ![image](https://user-images.githubusercontent.com/45002654/148334889-f702377e-a432-4166-b415-a0210b0a3d76.png) |
| --- |

Some more tests are still expected:
- The pure functions in the Redux code need testing
- Integration tests that ensure different parts of the app work well together should also be included.

Please consider adding more tests to the app.

rloterh replied:
This issue is well resolved! ✔️

Lameck1 reviewed (changes requested):
Hi, @ShoiraTa, 

So far so good 👍🏼. You've done such a splendid job on this React project.

### Some highlights
* Correct gitflow practice ✔️ 
* Good commit messages ✔️ 
* Proffessional README ✔️ 
* App deployed and available online ✔️ 
* In the project:
   * Good usage of the react-router-dom package ✔️ 
   * Clean codebase ✔️ 
   * Redux store and redux logic properly implemented ✔️  
   * All expected features of the MVP functioning as expected ✔️
   
We however have some issues to work on before consider working on this project complete.


## Required Changes ♻️
- [ ] _Check the comments under the review._
- [ ] In your video, please highlight and talk about a piece of code you found interesting. This part was not evident in the presentation.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ShoiraTa committed:
Add more tests

ShoiraTa commented:
I have added more tests includung pure function tests and integration tests:
![image](https://user-images.githubusercontent.com/77038610/148353694-56deb438-e05b-4e6b-9d97-49248778d252.png)

Updated the video and highlighted and a piece of code I found interesting

rloterh commented on README.md:
```diff
@@ -17,15 +17,17 @@ In this project, I was working with the real live data from the CORONAVIRUS DATA
+- [Netlify](https://wizardly-aryabhata-28f609.netlify.app/)
+
+
 
```
- [ ] Kindly change the header title _Authors_ to _Author_, as this project is done solely. This will make your ReadMe appear more professional.

ShoiraTa replied:
Hello @rloterh , thank you for your comment, however I believe that this requirement should be optional, because  It was not mentioned by the previous reviewer. The previous reviewer requested to add more tests and update the video.

peterrobert replied:
This change should have been optional since it wasn't requested by the first reviewer.

rloterh commented on README.md:
```diff
@@ -17,15 +17,17 @@ In this project, I was working with the real live data from the CORONAVIRUS DATA
 - npm
-- Heroku deployment
+- Netlify deployment
 
```
Your _ReadMe_ needs some improvements so as to make it appear more professional. Kindly note that _ReadMe_ is mostly the first place that visitors of this project's repository will go and therefore it has to be very detailed and professional. Also, it needs to be done well with respect to this capstone project,  which will likely feature on your portfolio. Therefore, kindly ensure to improve upon these aspects of your ReadMe:
- [ ] _**Project description**_, _how to **set up**_, _**run** and **test** this project_ **should be detailed, clear, and well** organized.
- [ ] The _code_ aspects in each procedure should be quoted with **backticks**.
- [ ] Include necessary screenshots, project demo links where applicable so as to give a birds-eye view of your project
- [ ] All _typos_ **should be corrected**. You can crosscheck by installing the [Grammarly plugin](https://www.grammarly.com/)

ShoiraTa replied:
@rloterh Could you please be more specific regarding the changes, I went through my readme several times but cannot find typos, and I already included 2 screenshots and instructions  about how to set up, run and test in Getting Started section.

ShoiraTa replied:
@rloterh  maybe you did not open readme file itself? you can check it here https://github.com/ShoiraTa/covid-cases-stats/blob/8c141760b16a417151ef8ccb25c64b09bcce59a6/README.md

peterrobert replied:
This change should have been optional

rloterh commented on src/components/Countries.js:
```diff
@@ -0,0 +1,108 @@
+import SingleCountry from './SingleCountry';
+import Navbar from './Navbar';
+
+function Countries() {
```
- [ ] OPTIONAL
It is a nice idea to prefer _arrow functions_ over normal functions, so as to keep your _function_ syntax simpler and cleaner.

rloterh commented on src/components/Regions.js:
```diff
@@ -0,0 +1,89 @@
+/* eslint-disable max-len */
```
- [ ] You have **disabled linter check** on your code. Please enable **all** linter check on your code to ensure that your code is quality and robust.

peterrobert replied:
This change should have been optional

rloterh reviewed (changes requested):
Hi Shoira, 

You did great on this project, and for making the changes requested in the previous review(s)! 👏👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  


## Required Changes ♻️
_Kindly check the comments below, and please take your time to carefully improve aspects outlined in the previous code review(s), and carefully check to see that the changes are working as intended._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider the comments marked **[OPTIONAL]** under this review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 3 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ShoiraTa commented:
@rloterh  I am very thankful to your comments, but I didn't find any required comment that was previously mentioned by first reviewer, thus, all the listed comments had to be **optional**, or am I wrong about this?

ShoiraTa committed:
Update Readme

ShoiraTa committed:
Delete disabled linter

peterrobert reviewed (approved):
Hi @ShoiraTa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
